python/tchannel/frame.py

```diff
@@ -1,101 +1,62 @@
 from __future__ import absolute_import
 
 import logging
+from collections import namedtuple
 
-from .context import Context
-from .exceptions import ProtocolException
-from .mapping import get_message_class
-from .parser import read_number
-from .parser import write_number
+from . import rw
+from .io import BytesIO
 
 log = logging.getLogger('tchannel')
 
 
-class Frame(object):
-    """Perform operations on a single TChannel frame."""
-    SIZE_WIDTH = 2
-    ID_WIDTH = 4
-    TYPE_WIDTH = 1
-    FLAGS_WIDTH = 1
-    PRELUDE_SIZE = 0x10  # this many bytes of framing before payload
+FrameHeader = namedtuple('FrameHeader', 'message_type message_id')
+Frame = namedtuple('Frame', 'header payload')
 
-    # 8 bytes are reserved
-    RESERVED_WIDTH = 8
-    RESERVED_PADDING = b'\x00' * RESERVED_WIDTH
 
-    BEFORE_ID_WIDTH = 1
-    BEFORE_ID_PADDING = b'\x00' * BEFORE_ID_WIDTH
+class FrameReadWriter(rw.ReadWriter):
 
-    def __init__(self, message, message_id):
-        self._message = message
-        self._message_id = message_id
+    # ReadWriter for Frame size
+    size_rw = rw.number(2)  # size:2
 
-    @classmethod
-    def decode(cls, stream, message_length=None):
-        """Decode a sequence of bytes into a frame and message.
+    # ReadWriter for FrameHeaders
+    header_rw = rw.instance(
+        FrameHeader,
+        ('message_type', rw.number(1)),         # type:1
+        ('_', rw.constant(rw.number(1), 0)),    # reserved:1
```

Comment:
Can you just define `rw.skip` as '_' or something so this is a little more explicit.